### DISCUSSION TITLE
dump is different in staging vs prod?

### DISCUSSION THREAD

**Discussion by shez1983** [spatie/laravel-db-snapshots#142] [Q&A]
so I am using a staging db (RDS, fixed) and prod db (Serverless) not sure if this should make any difference but is the only thing I can think of.

So I create the snapshots and load, on staging its fine, on prod it gives me an error.. strange thing i found is that the dumps look slightly different in both staging/prod ?
on staging it also includes primary key, but in prod it doesnt
on prod here is what it does:
```
INSERT INTO `addresses` VALUES
('Price Branch'...
````
and on staging:
```
INSERT INTO `addresses` VALUES
(1,'ContactProfile',1,NULL,'Price Branch',
```
actually having a second look it DOES include primary key but because it doesnt do :

insert into addresses (column names...) values (...)  sql thinks the first is an ID so errors out

googling led me to add 
```
 $command[] = '--complete-insert';'   
 ``` 
 in the getDumpCommand line 176/177 .. 
 
```vendor/spatie/db-dumper/src/Databases/MySql.php```

**Comment by keybrdist** [comment]
Thanks for posting this.  Adding the following to the source db connection worked for me:
```
'dump' => [
    'addExtraOption' => '--complete-insert',
],
```
